lib/http-revocations.js

```diff
@@ -0,0 +1,97 @@
+/*!
+ * Copyright (c) 2019-2020 Digital Bazaar, Inc. All rights reserved.
+ */
+'use strict';
+
+const {asyncHandler} = require('bedrock-express');
+const bedrock = require('bedrock');
+const brZCapStorage = require('bedrock-zcap-storage');
+const {config, util: {BedrockError}} = bedrock;
+const cors = require('cors');
+const database = require('bedrock-mongodb');
+const helpers = require('./helpers');
+
+bedrock.events.on('bedrock-express.configure.routes', app => {
+  const cfg = config['kms-http'];
+  const routes = {...cfg.routes};
+  routes.keystores = `${routes.basePath}/keystores`;
+  routes.keystore = `${routes.keystores}/:keystoreId`;
+  routes.revocations = `${routes.keystore}/revocations`;
+
+  // insert a revocation
+  app.options(routes.revocations, cors());
+  app.post(
+    routes.revocations,
+    // CORs is safe because revocation uses HTTP signatures + capabilities,
+    // not cookies
+    cors(),
+    // TODO: add zcap validator
+    //validate('bedrock-kms-http.zcap'),
+    asyncHandler(async (req, res) => {
+      // check revocation
+      const keystoreId = helpers.getKeystoreId(
+        {req, localId: req.params.keystoreId, routes});
+      const expectedTarget = `${keystoreId}/revocations`;
+      const expectedRootCapability = `${keystoreId}/zcaps/revocations`;
+      const {invoker} = await helpers.authorize({
+        req, expectedTarget, expectedRootCapability,
+        expectedAction: 'write'
+      });
+
+      // verify CapabilityDelegation before storing zcap
+      const capability = req.body;
+      const host = req.get('host');
+
+      let delegator;
+      try {
+        const results = await helpers.verifyDelegation(
+          {keystoreId, host, capability});
+        ({delegator} = results[0].purposeResult);
+        delegator = typeof _delegator === 'string' ? delegator : delegator.id;
+      } catch(e) {
+        throw new BedrockError(
+          'The provided capability delegation is invalid.',
+          'DataError', {
+            httpStatusCode: 400,
+            public: true,
+            message: e.message
+          }, e);
+      }
+      // ensure that the invoker of the write capability is the delegator
+      // of the capability to be revoked
+      if(delegator !== invoker) {
+        throw new BedrockError('Permission denied.', 'NotAllowedError');
+      }
+      await brZCapStorage.revocations.insert({delegator, capability});
+      res.status(204).end();
+    }));
+
+  app.get(
```

Comment:
Went to make changes to this API and noticed that it's not being used at all in the current flow in the test.  Can we drop this API entirely?

I believe we decided that for housekeeping purposed, we are marking stored delegations as 'deleted' in the delegations collection which is now in a EDV?  Therefore if a user would want to see revoked capabilities, they would look there.  Is there a current use case for this HTTP API for getting revocations?  We use the node API for checking revocations on the back-end.

Reply:
@mattcollier -- let's comment it out for now and add it when we need it.

Reply:
I'll be removing it as it is not a proper get API (using params) or a proper query API.